part2-장현태-sprint6/sprint-mission-6.sql

```diff
@@ -0,0 +1,215 @@
+        2025-03-03 |          133 | 2287.8999996185303
+        2025-03-04 |          144 |  2444.300001144409
+        2025-03-05 |          140 |  2350.650005340576
+        ```
+*/
+SELECT o.date, SUM(od.quantity) AS total_orders, SUM(od.quantity*p.price) AS total_amount
+FROM orders o
+JOIN order_details od ON o.id=od.order_id
+JOIN pizzas p ON od.pizza_id = p.id
+WHERE date >= '2025-03-01'
+  AND date < '2025-03-31'
+GROUP BY o.date
+ORDER BY o.date ASC;
+
+/*
+    3. `order`의 `id`가 78에 해당하는 주문 내역들을 조회합니다. 주문 내역에서 각각 주문한 피자의 이름을 `pizza_name`, 피자의 크기를 `pizza_size`, 피자 가격을 `pizza_price`, 수량을 `quantity`, 각 주문 내역의 총 금액을 `total_amount` 라는 이름으로 조회해 주세요.
+        
+        출력 예시:
+        
+        ```sql
+        The Thai Chicken Pizza      | S          |       12.75 |        1 |              12.75
+        The Big Meat Pizza          | S          |          12 |        1 |                 12
+        The Classic Deluxe Pizza    | S          |          12 |        1 |                 12
+        The Italian Capocollo Pizza | M          |          16 |        1 |                 16
+        The Spicy Italian Pizza     | L          |       20.75 |        3 |              62.25
+        The Four Cheese Pizza       | L          |       17.95 |        1 | 17.950000762939453
+        ```
+*/
+SELECT pt.name AS pizza_name, p.size AS pizza_size, p.price AS pizza_price,  od.quantity AS quantity, (p.price * od.quantity) AS total_amount 
+FROM orders o
+JOIN order_details od ON o.id=od.order_id
+JOIN pizzas p ON od.pizza_id = p.id
+JOIN pizza_types pt ON p.type_id = pt.id
+WHERE o.id = 78;
+
+/*    
+    4. `order_details`와 `pizzas` 테이블을 JOIN해서 피자 크기별(S, M, L) 총 수익을 계산하고, 크기별 수익을 출력하세요.
+        
+        출력 예시:
+        
+        ```sql
+        L    |  375318.7010040283
+        M    |          249382.25
+        S    | 178076.49981307983
+        XL   |              14076
+        XXL  | 1006.6000213623047
+        ```
+*/
+SELECT p.size,SUM(od.quantity*p.price)
+FROM pizzas p
+JOIN order_details od ON od.pizza_id = p.id
+GROUP BY p.size
+ORDER BY p.size;
+
+/*    
+    5. `order_details`, `pizzas`, `pizza_types` 테이블을 JOIN해서 각 피자 종류의 총 수익을 계산하고, 수익이 높은 순서대로 출력하세요.
+        
+        출력 예시:
+        
+        ```sql
+        The Thai Chicken Pizza                     |           43434.25
+        The Barbecue Chicken Pizza                 |              42768
+        The California Chicken Pizza               |            41409.5
+        The Classic Deluxe Pizza                   |            38180.5
+        The Spicy Italian Pizza                    |           34831.25
+        The Southwest Chicken Pizza                |           34705.75
+        The Italian Supreme Pizza                  |           33476.75
+        ```
+*/
+
+SELECT pt.name , SUM(od.quantity*p.price)
+FROM pizza_types pt
+JOIN pizzas p ON pt.id = p.type_id
+JOIN order_details od ON od.pizza_id = p.id
+GROUP BY pt.name
+ORDER BY SUM(od.quantity*p.price) DESC;
```

Comment:
총 수익에 별칭을 줘서 order_by에 활용할 수 있을 것 같습니다.
```suggestion
SELECT 
  pt.name , 
  SUM(od.quantity*p.price) as total_revenue
FROM pizza_types pt
JOIN pizzas p ON pt.id = p.type_id
JOIN order_details od ON od.pizza_id = p.id
GROUP BY pt.name
ORDER BY total_revenue DESC;
```